File: functions/processors/dive.js
const admin = require('firebase-admin');
const moment = require('moment');

const DiveProcessor = function (storageController) {

    this.storageController = storageController;

    const save = function* (dive, firestoreDoc) {
        const data = {
            startPoint: new admin.firestore.GeoPoint(dive.startLat, dive.startLong),
            endPoint: new admin.firestore.GeoPoint(dive.endLat, dive.endLong),
            sensorId: dive.sensorId,
            startTime: moment(dive.startTime).toDate(),
            endTime: moment(dive.endTime).toDate(),
            stringExample: 'path to GCS data'
        };

        console.log(moment().toDate());

    };

    return {
        save
    };
};

module.exports = DiveProcessor;

File: server/types/dive.go
package types

import (
	"time"
)

// Dive is a dive
type Dive struct {
	SensorID   string        `json:"sensorId" firestore:"sensorId,omitempty"`
	StartTime  time.Time     `json:"startTime"  firestore:"startTime,omitempty"`
	EndTime    time.Time     `json:"duration"  firestore:"endTime,omitempty"`
	StartPoint GeoPoint      `json:"startPoint" firestore:"startPoint,omitempty"`
	EndPoint   GeoPoint      `json:"endPoint" firestore:"endPoint,omitempty"`
	SensorData []*SensorData `json:"sensorData" firestore:"sensorData,omitempty"`
}
